presto-main/src/main/java/com/facebook/presto/util/MergeSortedPages.java

```diff
@@ -102,44 +102,42 @@ private MergeSortedPages() {}
     {
         LocalMemoryContext memoryContext = aggregatedMemoryContext.newLocalMemoryContext(MergeSortedPages.class.getSimpleName());
         PageBuilder pageBuilder = new PageBuilder(outputTypes);
-        return pageWithPositions.transform(pageWithPositionOptional -> {
-            if (yieldSignal.isSet()) {
-                return ProcessorState.yield();
-            }
-
-            boolean finished = !pageWithPositionOptional.isPresent();
-            if (finished && pageBuilder.isEmpty()) {
-                memoryContext.close();
-                return ProcessorState.finished();
-            }
+        return pageWithPositions
+                .yield(yieldSignal::isSet)
```

Comment:
Lines shifted...
this and https://github.com/prestodb/presto/pull/11764/commits/d3b66832c7158bb21c98e330c10c629445bf034d#diff-93db0a19be8a4b1b5d770cfd3d1e9350L106 are the only actual changes in this file.